climada/engine/uncertainty/base.py

```diff
@@ -0,0 +1,771 @@
+                " a surrogate model https://www.uqlab.com/")
+
+        ncpus = pool.ncpus if pool else 1
+        total_time = self.n_samples * time_one_run / ncpus
+        
+        return total_time
+
+
+    def calc_sensitivity(self, salib_method='sobol', method_kwargs=None):
+        """
+        Compute the sensitivity indices using SALib. Prior to doing this
+        sensitivity analysis, one must compute the distribution of the output
+        metrics values (i.e. self.metrics is defined) for all the parameter 
+        samples (rows of self.samples_df).
+        
+        According to Wikipedia, sensitivity analysis is “the study of how the
+        uncertainty in the output of a mathematical model or system (numerical
+        or otherwise) can be apportioned to different sources of uncertainty
+        in its inputs.” The sensitivity of each input is often represented by
+        a numeric value, called the sensitivity index. Sensitivity indices
+        come in several forms:
+
+        First-order indices: measures the contribution to the output variance
+        by a single model input alone.
+        Second-order indices: measures the contribution to the output variance
+        caused by the interaction of two model inputs.
+        Total-order index: measures the contribution to the output variance
+        caused by a model input, including both its first-order effects
+        (the input varying alone) and all higher-order interactions.
+
+
+        Parameters
+        ----------
+        salib_method : str
+            sensitivity analysis method from SALib.analyse
+            Possible choices:
+                'fast', 'rbd_fact', 'morris', 'sobol', 'delta', 'ff'
+            The default is 'sobol'.
+            Note that in Salib, sampling methods and sensitivity analysis
+            methods should be used in specific pairs.
+            https://salib.readthedocs.io/en/latest/api.html
+        method_kwargs: dict(), optional
+            Keyword arguments of the chosen SALib analyse method.
+            The default is to use SALib's default arguments.
+        Returns
+        -------
+        sensitivity_dict : dict
+            Dictionnary of the sensitivity indices. Keys are the
+            metrics names, values the sensitivity indices dictionnary
+            as returned by SALib.
+
+        """
+        
+        check_salib(self.sampling_method, salib_method)
+        
+        #To import a submodule from a module use 'from_list' necessary
+        #c.f. https://stackoverflow.com/questions/2724260/why-does-pythons-import-require-fromlist
+        method = getattr(
+            __import__('SALib.analyze',
+                       fromlist=[salib_method]
+                       ),
+            salib_method
+            )
+        
+        #Certaint Salib method required model input (X) and output (Y), others
+        #need only ouput (Y)
+        salib_kwargs = method.analyze.__code__.co_varnames
```

Comment:
I think I understand what you are doing, i.e., getting the full set of possible kwargs for the selected salib method (right?); but as this is not common (at least it is not for me) maybe it is good to say it explicitely in the comment just above.

Reply:
I added a comment in 6c66e6729cf297e478eb6501d835ba3bdc26c0e5